apps/screenshot-diff/src/computeScreenshotDiff.ts

```diff
@@ -123,21 +124,26 @@ export const computeScreenshotDiff = async (
     });
 
     if (diffResult.score > 0) {
+      const diffKey = await hashFile(diffResult.filepath);
+      const duplicateFile = await File.query().findOne({ key: diffKey });
       const diffImage = new S3ImageFile({
         s3,
         bucket: bucket,
         filepath: diffResult.filepath,
+        ...(duplicateFile && { key: duplicateFile.key }),
       });
       const key = await diffImage.upload();
```

Comment:
I think you will upload it anyway. Not good I think.

Reply:
Yes and no. I won't be re-uploaded but it will throw an error. You are right, it's not good enough

![Capture d’écran 2023-09-23 à 19 19 59](https://github.com/argos-ci/argos/assets/15954562/88f5c929-b4a2-4d28-9e96-1ae87e4d31b8)